rcsb/app/tests-client/testClientUtils.py

```diff
@@ -9,253 +9,803 @@
 __email__ = "james.smith@rcsb.org"
 __license__ = "Apache 2.0"
 
+import subprocess
 import logging
 import os
 import platform
 import resource
-import time
 import unittest
 import shutil
 
 # requires server
 
-from rcsb.app.file import __version__
 from rcsb.app.file.ConfigProvider import ConfigProvider
 from rcsb.utils.io.FileUtil import FileUtil
 from rcsb.utils.io.LogUtil import StructFormatter
 from rcsb.app.client.ClientUtils import ClientUtils
+from rcsb.app.file.PathProvider import PathProvider
 
-logging.basicConfig(level=logging.INFO, format="%(asctime)s [%(levelname)s]-%(module)s.%(funcName)s: %(message)s")
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s]-%(module)s.%(funcName)s: %(message)s",
+)
 logger = logging.getLogger()
 root_handler = logger.handlers[0]
 root_handler.setFormatter(StructFormatter(fmt=None, mask=None))
 logger.setLevel(logging.INFO)
 
 
 class ClientTests(unittest.TestCase):
+    # comment out if running gunicorn or uvicorn
+    # runs only once
+    @classmethod
+    def setUpClass(cls):
+        subprocess.Popen(
+            ["uvicorn", "rcsb.app.file.main:app"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+        )
+
+    # comment out if running gunicorn or uvicorn
+    # runs only once
+    @classmethod
+    def tearDownClass(cls):
+        os.system(
+            "pid=$(ps -e | grep uvicorn | head -n1 | awk '{print $1;}';);kill $pid;"
+        )
```

Comment:
Since we're not using TestClient, the testClientUtils file requires a server. One would typically run gunicorn/uvicorn first, then run the testClientUtils file. However, that's not possible on Azure, so the setUpClass/tearDownClass functions were added for that purpose. You should use one or the other.